### ISSUE TITLE
CNTK to ONNX converter bug - inception-v3 model

### ISSUE THREAD

**Issue by JohnCraigPublic** [microsoft/onnxruntime#3428]
**Describe the bug**
Try to export an inception-v3 model in CNTK to ONNX yields the error message:

Error: two nodes with same node name (model.model.pool_2_Output_reshape_to_model.model.pool_2_Output_reshaped_for_conv).

**Urgency**
Need something soon.  April 15th?

**System information**
- OS Platform  = Windows 10
- ONNX Runtime Nuget 1.20
- VS2017 using C++
- CNTK   rc0.dev20200331

**To Reproduce**
The CNTK model is here.   Just try converting it to ONNX.  I was doing so from C++
https://drive.google.com/open?id=1Z01D5zrbBiwvu4EQEqNK__P7FH2VV26Q

**Expected behavior**
Should produce working ONNX model

**Additional context**
The inception-v3 model only differs from published versions in that it expects single-channel (gray) images.   It is an image classifier for 112 classes.   The CNTK model works fine,  Need to get it out to ONNX.

**Comment by KeDengMS** [comment]
If you have time pressure, could the PyTorch version help?https://pytorch.org/docs/stable/torchvision/models.html#torchvision.models.inception_v3

**Comment by JohnCraigPublic** [comment]
KeDengMS - thanks - I've grabbed the Inception model available in Keras/Tensorflow and have gotten that to work, including the conversion to ONNX.  So I will continue down that path.  However, I still have interest in being able to convert a few years worth of CNTK models (based on Inception) to ONNX if possible.

**Comment by liqunfu** [comment]
John and Ke,


I have a draft fix now. The root cause is that John is still using CNTK brain script which automatically adds sequence axis. Sequence axis is not supported in ONNX so I need to wrap some ops (pooling, conv, BN) with reshape ops.



To complete the work we need to follow the same process we did just 1 or 2 weeks ago. It involves other resources to test and build the nuget package. So let us collectively decide whether and how to proceed.



@JohnCrag in the mean time please migrate your models to CNTK V2 or better to Pytorch. ONNX feature in Pytorch is actively supported.



Thanks,

Liqun


________________________________
发件人: John Craig <notifications@github.com>
发送时间: 2020年4月14日 1:31
收件人: microsoft/onnxruntime <onnxruntime@noreply.github.com>
抄送: Subscribed <subscribed@noreply.github.com>
主题: Re: [microsoft/onnxruntime] CNTK to ONNX converter bug - inception-v3 model (#3428)


KeDengMS - thanks - I've grabbed the Inception model available in Keras/Tensorflow and have gotten that to work, including the conversion to ONNX. So I will continue down that path. However, I still have interest in being able to convert a few years worth of CNTK models (based on Inception) to ONNX if possible.

―

**Comment by JohnCraigPublic** [comment]
liqunfu - how can I "migrate my models to CNTK V2" -- by this do you mean to recompute them not using Brainscript?  Or do you mean that there is a way I could convert my models to CNTK V2 without re-training them from scratch?

**Comment by liqunfu** [comment]
I mean to rewrite you models in V2 or Pytorch and retrain them.

________________________________
发件人: John Craig <notifications@github.com>
发送时间: 2020年4月15日 13:05
收件人: microsoft/onnxruntime <onnxruntime@noreply.github.com>
抄送: liqunfu <liqun_fu@hotmail.com>; Comment <comment@noreply.github.com>
主题: Re: [microsoft/onnxruntime] CNTK to ONNX converter bug - inception-v3 model (#3428)


liqunfu - how can I "migrate my models to CNTK V2" -- by this do you mean to recompute them not using Brainscript? Or do you mean that there is a way I could convert my models to CNTK V2 without re-training them from scratch?

―

**Comment by JohnCraigPublic** [comment]
liqunfu - OK, I am starting down that path, using Keras/Tensorflow.

**Comment by JohnCraigPublic** [comment]
liqunfu - is there a new CNTK with your fix so I can export inception model as ONNX ?

**Comment by JohnCraigPublic** [comment]
liqunfu - last time you made a fix to ONNX converter you sent me a ZIP file so I would update my NuGet packages with your latest -- can you do that again now?  For your latest change so I can convert an Inception model from CNTK to ONNX?  Thanks.

**Comment by Zengyajunbit** [comment]
I met the same problem. Has this problem been solved？

**Comment by JohnCraigPublic** [comment]
I see that on the nightly CNTK page there is mention of  2.8-rc0.dev20200502

But in VS2017, the NuGet Package Manager, with "include prereleases" checked does not offer it.  How to get it?  Perhaps it solves the problem that I've been waiting months for.  Very frustrating.

**Comment by JohnCraigPublic** [comment]
I just installed nightly package ...0509 in hope that this bug might be fixed.  But no,  still cannot convert an Inception-V3 model from CNTK to ONNX!    Catch this exception:

Error: two nodes with same node name (model.model.pool_2_Output_reshape_to_model.model.pool_2_Output_reshaped_for_conv).
